compiler/rustc_macros/src/diagnostics/utils.rs

```diff
@@ -422,55 +424,53 @@ pub(super) enum AllowMultipleAlternatives {
 /// `#[suggestion*(code("foo", "bar"))]` attribute field
 pub(super) fn build_suggestion_code(
     code_field: &Ident,
-    meta: &Meta,
+    nested: ParseNestedMeta<'_>,
     fields: &impl HasFieldMap,
     allow_multiple: AllowMultipleAlternatives,
 ) -> TokenStream {
-    let values = match meta {
-        // `code = "foo"`
-        Meta::NameValue(MetaNameValue { lit: syn::Lit::Str(s), .. }) => vec![s],
-        // `code("foo", "bar")`
-        Meta::List(MetaList { nested, .. }) => {
+    let values = match (|| {
```

Comment:
Could you use a try block and a separate variable here? The `match (|| {` scares me a lot. Maybe it would even make sense to create an entirely separate function for the parsing part.